src/ServiceControlInstaller.Engine/Database/DatabaseMigrations.cs

```diff
@@ -0,0 +1,67 @@
+﻿namespace ServiceControlInstaller.Engine.Database
+{
+	using System;
+	using System.Diagnostics;
+	using System.IO;
+	using ServiceControlInstaller.Engine.Instances;
+
+	internal class DatabaseMigrations
+	{
+		public static void RunDatabaseMigrations(IServiceControlInstance instance, Action<string> updateProgress)
+		{
+			RunDataMigration(updateProgress, instance.InstallPath,
+				Constants.ServiceControlExe,
+				instance.Name);
+		}
+
+		static void RunDataMigration(Action<string> updateProgress, string installPath, string exeName, string serviceName)
+		{
+			var processStartupInfo = new ProcessStartInfo
+			{
+				CreateNoWindow = true,
+				UseShellExecute = false,
+				FileName = Path.Combine(installPath, exeName),
+				Arguments = $"--database --serviceName={serviceName}",
+				WorkingDirectory = installPath,
+				RedirectStandardOutput = true,
+				RedirectStandardError = true
+			};
+
+			var attempts = 0;
+			Process p;
+			var updatingSchema = false;
+			do
+			{
+				attempts++;
+
+				p = Process.Start(processStartupInfo);
+				if (p == null)
+				{
+					throw new Exception($"Attempt to launch {exeName} failed.");
+				}
+
+				p.OutputDataReceived += (sender, eventArgs) =>
+				{
+					if (eventArgs.Data.StartsWith("Updating schema from version"))
+					{
+						updatingSchema = true;
+						updateProgress(eventArgs.Data.Remove(':'));
+					}
+					else if (updatingSchema && eventArgs.Data.StartsWith("OK"))
+					{
+						updatingSchema = false;
+						updateProgress(string.Empty);
+					}
+				};
+
+				p.WaitForExit((int)TimeSpan.FromMinutes(20).TotalMilliseconds);
```

Comment:
We need some sort of max time for any migration. I just pulled 20 minutes out of thin air. Thoughts on how long we should wait?